### ISSUE TITLE
display of torrent contents gets reset during download

### ISSUE THREAD

**Issue by as-muncher** [BiglySoftware/BiglyBT#2459]
Java 1.8.0_202 (64 bit)
  Oracle Corporation
c:\program files\biglybt\jre

SWT v4942r22, win32, zoom=100, dpi=120
Windows 10 v10.0, amd64 (64 bit)
B2.9.0.0/4 az3

So, when I am downloading a torrent, and it has a section for screenshots, if I click the triangle so that the screenshots listing gets compressed to just the mention of the screenshots folder, I think it's when a file in that torrent gets fully downloaded that the display gets reset.  All of a sudden, the screenshots directory is not collapsed anymore.  I'll collapse it again, and then a short while later, it shows again, maybe because another file has been downloaded.  Could you please fix this so that how I have it arranged will stay that way?

**Comment by as-muncher** [comment]
or if I've scrolled down to look at the bottom contents of the torrent, a file finishes downloading and the display is reset, and I'm back at the top of the contents.

**Comment by parg** [comment]
Try B35

**Comment by as-muncher** [comment]
ok thanks

**Comment by as-muncher** [comment]
Java 1.8.0_202 (64 bit)
  Oracle Corporation
c:\program files\biglybt\jre

SWT v4942r22, win32, zoom=100, dpi=120
Windows 10 v10.0, amd64 (64 bit)
B2.9.0.1_B36/4 az3

This is happening, still, but with highlighting.  If I have certain files in my torrent highlighted, that gets reset.

**Comment by parg** [comment]
So your point is that the tree expansion state now gets saved correctly but you have a different issue in that the selection state doesn't?

**Comment by as-muncher** [comment]
@parg, Ok, I will make a new issue.

**Comment by as-muncher** [comment]
> So your point is that the tree expansion state now gets saved correctly but you have a different issue in that the selection state doesn't?

and thank you for fixing the issue.  :)  Thanks for all the work you do.

**Comment by as-muncher** [comment]
Wow, this is better than I expected.  BiglyBT remembers the various files I've selected in my torrent after I've moved on to another torrent.  However, I still notice that the display of the contents of the torrent are changing to display the beginning of the torrent contents even though I'm currently looking at certain selected files or scrolled down, after one more file in the torrent has downloaded.  But the selections still remain, which is fantastic - thank you.
